ee/tabby-db/migrations/0028_gitlab-provider.up.sql

```diff
@@ -0,0 +1,19 @@
+CREATE TABLE gitlab_repository_provider(
+    id INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,
+    display_name TEXT NOT NULL,
+    access_token TEXT
```

Comment:
add a field `connected` to present its status.

1. When it's first created, default to `false`.
2. When cron job complete successfully, set it to `true`.
3. When cron job has error and access token got reset, set it to `false`.

The same logic should be added to github repository provider as well, but we can do it in a separate PR.

Reply:
TAB-634